### PR TITLE
Run ktlint as part of Gradle build

### PR DESCRIPTION
**Pull request by MikkCZ**
No need to install ktlint locally anymore, Gradle can do it all.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ Contribute
 ----------
 
 The Kotlin sources in this project use `ktlint <https://ktlint.github.io/>`__.
-Please run ``ktlint -F`` before submitting patches.
+To resolve linting errors, run ``./gradlew ktlintFormat`` before submitting patches.
```
Comment (Pike) [inline README.rst:20]: This section would need to go into the README in `fluent.syntax` now. There's no top-level gradlew. On purpose, as we might add a java 7 integration example, and that would require a different gradle and dependencies that we use elsewhere.

**Comment by MikkCZ** [comment]
The ktlint version used by the plugin is defined by the line `version.set("0.37.2")`.

File: README.rst (hunk 1 above)
Comment (MikkCZ) [reply README.rst]: Updated.

A note about Java world - compared to Python 2/3 for example, it's extremely compatible. So if you decide to support Java 7 and not just 8+, just compile it with JDK 7, and that's it. Older JVMs won't understand newer classfiles, but newer JVMs do understand older clasfiles. If you really want to, you can distribute packaged with classfiled build by different version of Java, but there is barely any benefit there. Unless there are some extreme cases, which I cannot come up with from the top of my head, there is no need to have multiple projects, setups, builds or even artifacts to support multiple versions of Java.

Comment (Pike) [reply README.rst]: AFAICT, both gradle 6 and junit5 refuse to run on java 7. Gradle can shell out to 7 to run tests, but then you can't use junit5. I want junit5 for dynamic tests, though.

Comment (MikkCZ) [reply README.rst]: Is there actually demand for Java 7 support? Most of the Java world I know moved to 8 or 11, which are the LTS versions with significantly longer lifespan ahead (and compared to Java 7, also some very interesting features when speaking about Java language).

File: .github/workflows/fluent.syntax.yml [hunk 2]
```diff
@@ -7,7 +7,8 @@ on:
     branches: [ master ]
 
 jobs:
-  build:
+  test:
+    name: run tests
```
Comment (Pike) [inline .github/workflows/fluent.syntax.yml:11]:
```suggestion
    name: build & test
```

File: .github/workflows/fluent.syntax.yml [hunk 3]
```diff
@@ -17,17 +18,16 @@ jobs:
         java-version: 1.8
     - name: Build and check
       working-directory: ./fluent.syntax
-      run: ./gradlew check
+      run: ./gradlew test
   lint:
+    name: lint Kotlin
```
Comment (Pike) [inline .github/workflows/fluent.syntax.yml:23]:
```suggestion
    name: lint
```

File: .github/workflows/fluent.syntax.yml [hunk 4]
```diff
@@ -17,17 +18,16 @@ jobs:
...
-      working-directory: ./fluent.syntax
-      run: ktlint
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Build and check
+        working-directory: ./fluent.syntax
+        run: ./gradlew ktlintCheck
```
Comment (Pike) [inline .github/workflows/fluent.syntax.yml:33]: Align the indention here

```suggestion
    - uses: actions/checkout@v2
    - name: Set up JDK 1.8
      uses: actions/setup-java@v1
      with:
        java-version: 1.8
    - name: Build and check
      working-directory: ./fluent.syntax
      run: ./gradlew ktlintCheck
```

**Pull request merged**